cms/core/static/js/blocks/readonly-struct-block.js

```diff
@@ -0,0 +1,38 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+class ReadonlyStructBlockDefinition extends window.wagtailStreamField.blocks.StructBlockDefinition {
+  frozenHelpText = '';
+
+  hiddenFields = [];
+
+  render(placeholder, prefix, initialState, initialError) {
+    const block = super.render(placeholder, prefix, initialState, initialError);
+
+    const parent = `[data-streamfield-child]:has([id^="${prefix}"])`;
+
+    if (initialState.frozen) {
+      const helpText = document.querySelector(`${parent} .help`);
+      helpText.innerHTML = this.frozenHelpText;
+
+      const getActionButton = (action) =>
+        document.querySelector(`${parent} [data-streamfield-action="${action}"]`);
+
+      const deleteButton = getActionButton('DELETE');
+      deleteButton.style.display = 'none';
+      const duplicateButton = getActionButton('DUPLICATE');
+      duplicateButton.style.display = 'none';
+    }
+
+    const getFieldContainer = (field) =>
+      document.querySelector(`${parent} [data-contentpath="${field}"]:has(#${prefix}-${field})`);
+
+    const frozenCheckbox = getFieldContainer('frozen');
+    frozenCheckbox.style.display = 'none';
```

Comment:
suggestion: what happens if the frozen field is not there?

Granted we use it in focussed way. Perhaps add a docblock to the class outlining that the StructBlock must have help text and a `frozen = blocks.BooleanBlock(required=False, default=False)` block too?